### PR TITLE
ss-106: fix auth redirect

### PR DESCRIPTION
**Pull request by Evcat6**

### CODE REVIEW

File: frontend/src/app/app.tsx [hunk 1]
```diff
@@ -1,23 +1,34 @@
...
 
     useEffect(() => {
         if (isRoot) {
             void dispatch(userActions.loadAll());
         }
-    }, [isRoot, dispatch]);
+    }, [dispatch, isRoot]);
+
+    useEffect(() => {
+        if (!user && token) {
```
Comment (nikita-remeslov) [inline frontend/src/app/app.tsx:28]: we don't care about token. If we don't have token we will get error

File: frontend/src/bundles/auth/store/actions.ts [hunk 2]
```diff
@@ -36,6 +39,7 @@ const signIn = createAsyncThunk<
     void storage.set(StorageKey.TOKEN, token);
 
     await dispatch(loadUser());
+    return { token };
```
Comment (nikita-remeslov) [inline frontend/src/bundles/auth/store/actions.ts:42]: why do we need a return token?

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/app/app.tsx (hunk 1 above)
Comment (Evcat6) [reply frontend/src/app/app.tsx]: Resolved in [62a6191](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/120/commits/62a6191e0a0cc77a168d461945b777155b7ef161)

File: frontend/src/bundles/auth/store/actions.ts (hunk 2 above)
Comment (Evcat6) [reply frontend/src/bundles/auth/store/actions.ts]: Resolved in [aa800ed](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/120/commits/aa800edc5e4cac6b991472363a64b9834428bccc)

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**